curs_main.c

```diff
@@ -2681,7 +2686,7 @@ int mutt_index_menu(void)
         }
         else
         {
-          mutt_error(_("Thread contains unread messages."));
+          mutt_error(_("Thread contains unread or flagged messages."));
           break;
```

Comment:
Ah, this isn't true, we're missing a test.

If I set `collapse_flagged = no`, then when I enter a folder, flagged messages aren't folded.
However, I can still manually collapse the thread even if it contains a flagged message.

With `collapse_unread = no`, the thread is open when I enter the folder and I **cannot** manually fold it.